aesara/tensor/nlinalg.py

```diff
@@ -823,6 +825,112 @@ def tensorsolve(a, b, axes=None):
     return TensorSolve(axes)(a, b)
 
 
+class MatMul(Op):
+    __props__ = ("dtype",)
+
+    def __init__(self, dtype=None):
+        self.dtype = dtype
+
+    def _get_output_shape(self, x1, x2, shapes, validate=False):
+        x1_shape, x2_shape = shapes
+
+        if x1.ndim == 1 and x2.ndim == 1:
```

Comment:
As per your comment, I don't find this necessarily off-putting. If you really want to avoid repeating conditionals you can always use dictionaries:

```python
cases = [
    {
        "case_fn": lambda x1, x2: x1.ndim==1 and x2.ndim ==1,
        "error_fn": lambda x1_shape, x2_shape: x1_shape[0] != x2_shape[0],
        "error_msg": "1d inputs must have the same length."
    }
]

if validate:
    for case in cases:
        x1_shape, x2_shape = shapes
        if case["cond_fn"](x1, x2) and case["error_fn"](x1_shape, x2_shape):
            raise ValueError(case["error_msg"])      
```

But it doesn't make the code any clearer imho. Repetition is not necessarily bad. Having said that you could return before all the conditionals if `validate == False`

Reply:
Yeah, it looks like trying to be clever here is not going to improve readability. I'll leave it as is.